ReactWindows/ReactNative/Common/AccessibilityHelper.cs

```diff
@@ -0,0 +1,608 @@
+        /// Must be called when a child is removed from <paramref name="parent"/>.
+        /// </summary>
+        /// <param name="parent">The element from which a child is removed.</param>
+        public static void OnChildRemoved(UIElement parent)
+        {
+            // Post on UI thread because the method requires that parent/child
+            // relationship is established but XAML does not set the relationship
+            // until the next loop after the child is added.
+            DispatcherHelpers.RunOnDispatcher(() =>
+            {
+                UpdateGeneratedNameHereAndUp(parent);
+            });
+        }
+
+        /// <summary>
+        /// Internal implementation of <see cref="OnChildAdded(UIElement, DependencyObject)"/>.
+        /// </summary>
+        /// <param name="parent"></param>
+        /// <param name="child"></param>
+        private static void OnChildAddedInternal(UIElement parent, DependencyObject child)
+        {
+            UpdateImportantForAccessibilityForAddedChild(parent, child);
+            UpdateGeneratedNameHereAndUp(parent);
+        }
+
+        /// <summary>
+        /// If an UIElement owning the <paramref name="peer"/> can be determined, it is returned,
+        /// otherwise returns null.
+        /// </summary>
+        /// <param name="peer"></param>
+        /// <returns></returns>
+        private static UIElement GetUIElementFromAutomationPeer(AutomationPeer peer)
+        {
+            UIElement ret = null;
+
+            switch (peer)
+            {
+                case FrameworkElementAutomationPeer feAutomationPeer:
+                    ret = feAutomationPeer.Owner;
+                    break;
+                case ItemAutomationPeer itemAutomationPeer:
+                    ret = itemAutomationPeer.Item as UIElement;
+                    break;
+                default:
+                    break;
+            }
+
+            return ret;
+        }
+
+        #region ImportantForAccessibility
+
+        /// <summary>
+        /// Sets the ImportantForAccessibility property for <paramref name="uiElement"/>.
+        /// It uses AutomationProperties.AccessibilityView to expose the element and its children
+        /// to narrator. ImportantForAccessibility value is stored as an attached property.
+        /// </summary>
+        /// <param name="uiElement">Element which ImportantForAccessibility property is set.</param>
+        /// <param name="importantForAccessibility">The new value of ImportantForAccessibility property.</param>
+        public static void SetImportantForAccessibility(UIElement uiElement, ImportantForAccessibility importantForAccessibility)
+        {
+            // Post on UI thread because the method requires that parent/child
+            // relationship is established but XAML does not set the relationship
+            // until the next loop after the child is added.
+            DispatcherHelpers.RunOnDispatcher(() =>
```

Comment:
>RunOnDispatcher [](start = 30, length = 15)

follow up on passing dispatcher here or scheduling all the work later on framework level

Reply:
This was addressed, dispatcher reliance was removed.

---
In reply to: [169482707](https://github.com/Microsoft/react-native-windows/pull/1645#discussion_r169482707) [](ancestors = 169482707)